packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -176,19 +175,25 @@ export class SummarizeResultBuilder {
 }
 
 /**
- * Errors type for errors hit during summary that may be retriable.
+ * Error type for errors hit during summarization that may be retriable.
  */
 export class RetriableSummaryError extends LoggingError {
 	public readonly canRetry = this.retryAfterSeconds !== undefined;
 	constructor(
 		message: string,
-		public readonly retryAfterSeconds?: number,
+		public readonly retryAfterSeconds: number | undefined,
+		public readonly retryCount: number | undefined,
 		props?: ITelemetryProperties,
 	) {
 		super(message, props);
 	}
 }
 
+/** Check retryCount property on RetriableSummaryError */
+export const getRetryCountFromError = (
```

Comment:
Is this that much better than just doing `error?.retryCount` inlined?  Or is there an implicit type conversion going on with the function param that you need?

Reply:
I simplified the whole logic, so this change is removed.